course5/homework04/homework04.cc

```diff
@@ -125,6 +135,19 @@ int main(int, char**)
 	Cout << "2 + 2 = " << 2 + 2 << "\n";
 	s << "hey" << Endl;
 	s << -2 << Endl;
-	yolo(s);
-	std::abort();
+	yolo(std::move(s));
+
+	Stream sToMove("another.txt");
+	Stream s2 = std::move(sToMove); // test move constructor
+	s2 << "\n :) \n" << Endl;
+
+	Stream s3("more.txt");
+	s3 = std::move(sToMove);  // test move assignment
+	s3 << "\n :( \n" << Endl;
```

Comment:
I still haven't implemented the "don't close the file if it's already a nullptr" bit, but this isn't segfaulting... I added printf to the destructor, and it says it's closing the file twice. Wut.  Does the `std::move` somehow magically keep both objects in scope? What am I doing wrong?